src/cudamatrix/cu-device.cc

```diff
@@ -216,7 +216,10 @@ void CuDevice::FinalizeActiveGpu() {
     active_gpu_id_ = act_gpu_id; // CuDevice::Enabled() is true from now on
     // Initialize the CUBLAS
     CU_SAFE_CALL(cublasCreate(&handle_));
-
+  #if HAVE_CUNN == 1
+    // Initialize CUDNN
+    CU_SAFE_CALL(cudnnCreate(&cudnn_));
```

Comment:
You'll need to place a macro like this in cu-common.h

#define CUDNN_SAFE_CALL(fun) \
{ \
  cudnnStatus_t ret; \
  if ((ret = (fun)) != CUDNN_STATUS_SUCCESS) { \
    KALDI_ERR << "cudnnStatus_t " << ret << " : \"" << cuddnnGetErrorString(ret) << "\" returned from '\
" << #fun << "'"; \
  } \
  cudaDeviceSynchronize(); \
}

and use it instead of CU_SAFE_CALL. The reason why is that cudnn's functions return a totally different type  (cudnnStatus_t) than CUDA itself (cudaError_t) to report errors. You can see the definition of cudaError_t here: http://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__TYPES.html#group__CUDART__TYPES_1gf599e5b8b829ce7db0f5216928f6ecb6

Meanwhile, the definition of cudaError_t is totally different. You can see that in cudnn.h. We need this so users get accurate error messages.